src/content/Form/Form.tsx

```diff
@@ -1,17 +1,36 @@
-import { ISectionData } from "../App/App.types"
-import "./Form.css"
-import { useContext } from "react"
-import { ModalDispatchContext, ModalPreset } from "../ModalLayer"
+import { ISectionData } from "../App/App.types";
+import "./Form.css";
+import { useContext, useState } from "react";
+import { ModalDispatchContext, ModalPreset } from "../ModalLayer";
+import SyncSavedSchedules from "../SyncSavedSchedules/SyncSavedSchedules";
 
 interface IProps {
-  newSection: ISectionData | null
-  sectionConflict: boolean
-  handleAddNewSection: () => void
-  handleCancel: () => void
+  sections: ISectionData[];
+  newSection: ISectionData | null;
+  sectionConflict: boolean;
+  handleAddNewSection: () => void;
+  handleCancel: () => void;
+  currentWorklist: number;
 }
 
 const Form = (props: IProps) => {
-  const dispatchModal = useContext(ModalDispatchContext)
+  const dispatchModal = useContext(ModalDispatchContext);
+  const [isSyncComponentVisible, setIsSyncComponentVisible] = useState(false);
+
+  const handleSyncClick = () => {
+    setIsSyncComponentVisible(true);
+  };
+
+  const closeSyncComponent = () => {
+    setIsSyncComponentVisible(false);
+  };
+
+  let sectionsForWorklist:ISectionData[] = []
+  for (const section of props.sections) {
```

Comment:
I feel like this logic probably be extracted up to `App.tsx` as a `currentWorklistSections` handler - it's probably useful elsewhere in the codebase as well.

Also, ESLint errors

Reply:
I like that idea. Went and extracted it all the way to ``util.ts`` as a utility function, ``filterSectionsByWorklist`` so that it can be called anywhere to filter without passing around extra props. For example, ``App.tsx`` now passes to ``Form.tsx`` ``sections={filterSectionsByWorklist(sections, currentWorklistNumber)}``